systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java

```diff
@@ -623,60 +600,56 @@ void testRemoveUserOperatorFromEntityOperator(ExtensionContext extensionContext)
         }
 
+        LOGGER.info("Recreate User Operator");
         KafkaResource.replaceKafkaResourceInSpecificNamespace(clusterName, k -> k.getSpec().getEntityOperator().setUserOperator(new EntityUserOperatorSpec()), namespaceName);
         //Waiting when EO pod will be recreated with UO
         DeploymentUtils.waitForDeploymentAndPodsReady(namespaceName, KafkaResources.entityOperatorDeploymentName(clusterName), 1);
 
-        //Checking that UO was created
-        kubeClient(namespaceName).listPodsByPrefixInName(namespaceName, KafkaResources.entityOperatorDeploymentName(clusterName)).forEach(pod -> {
-            pod.getSpec().getContainers().forEach(container -> {
-                assertThat(container.getName(), anyOf(
-                        containsString("topic-operator"),
-                        containsString("user-operator"),
-                        containsString("tls-sidecar"))
-                );
-            });
-        });
+        int expectedEOContainerCount = Environment.isKRaftModeEnabled() ? 1 : 3;
+        PodUtils.waitUntilPodContainersCount(namespaceName, KafkaResources.entityOperatorDeploymentName(clusterName), expectedEOContainerCount);
 
-        Instant endTime = Instant.now();
-        long duration = Duration.between(startTime, endTime).toSeconds();
-        assertNoCoErrorsLogged(namespaceName, duration);
-    }
+        LOGGER.info("Verify that Entity operator and all its component are correctly recreated");
+        // names of containers present in EO pod
+        List<String> entityOperatorContainerNames = kubeClient().listPodsByPrefixInName(namespaceName, KafkaResources.entityOperatorDeploymentName(clusterName))
+                .get(0).getSpec().getContainers()
+                .stream()
+                .map(Container::getName)
+                .toList();
 
-    @ParallelNamespaceTest
-    @KRaftNotSupported("EntityOperator is not supported by KRaft mode and is used in this test class")
-    void testRemoveUserAndTopicOperatorsFromEntityOperator(ExtensionContext extensionContext) {
-        final String namespaceName = StUtils.getNamespaceBasedOnRbac(namespace, extensionContext);
-        final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        assertThat("user-operator container is not present in EO", entityOperatorContainerNames.stream().anyMatch(name -> name.contains("user-operator")));
 
-        Instant startTime = Instant.now();
-        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(clusterName, 3).build());
+        // kraft does not support Topic Operator, removal of User Operator, and recreation of Entity Operator is all that can be tested here.
+        if (Environment.isKRaftModeEnabled()) {
+            Instant endTime = Instant.now();
+            long duration = Duration.between(startTime, endTime).toSeconds();
+            assertNoCoErrorsLogged(namespaceName, duration);
+            return;
```

Comment:
I do not understand why you check CO logs here?

Reply:
Checks for errors in CO log are done right after each test case so I am not sure if it makes sense here. We are only missing functionally to collect logs from previously restarted pods, so if you restart the CO it makes sense then.

Reply:
thanks, check really seems unnecessary.